Source/SwiftLintBuiltInRules/Rules/Style/VerticalWhitespaceClosingBracesRule.swift

```diff
@@ -14,52 +15,304 @@ struct VerticalWhitespaceClosingBracesRule: CorrectableRule, OptInRule {
+                    consecutiveNewlines: consecutiveNewlines,
+                    violationStartPosition: violationStartPosition,
+                    violationEndPosition: violationEndPosition
+                )
+                let result = processNewlines(
+                    count: count,
+                    bytesPerNewline: 2,
+                    context: &context
+                )
+                violationStartPosition = context.violationStartPosition
+                violationEndPosition = context.violationEndPosition
+                return result
+            case .spaces, .tabs:
+                return (consecutiveNewlines, piece.sourceLength.utf8Length)
+            default:
+                // Any other trivia breaks the sequence
+                violationStartPosition = nil
+                violationEndPosition = nil
+                return (0, piece.sourceLength.utf8Length)
+            }
         }
-    }
 
-    func correct(file: SwiftLintFile) -> Int {
-        let pattern = configuration.onlyEnforceBeforeTrivialLines ? self.trivialLinePattern : self.pattern
-        let violatingRanges = file.ruleEnabled(violatingRanges: file.violatingRanges(for: pattern), for: self)
-        guard violatingRanges.isNotEmpty else {
-            return 0
+        private func processNewlines(
+            count: Int,
+            bytesPerNewline: Int,
+            context: inout NewlineProcessingContext
+        ) -> (newlines: Int, positionAdvance: Int) {
+            var newConsecutiveNewlines = context.consecutiveNewlines
+            var totalAdvance = 0
+
+            for _ in 0..<count {
+                newConsecutiveNewlines += 1
+                // violationStartPosition marks the beginning of the first newline
+                // that constitutes an empty line (i.e., the second in a sequence of \n\n).
+                if newConsecutiveNewlines == 2 && context.violationStartPosition == nil {
+                    context.violationStartPosition = context.currentPosition.advanced(by: totalAdvance)
+                }
+                // violationEndPosition tracks the end of the last newline in any sequence of >= 2 newlines.
+                if newConsecutiveNewlines >= 2 {
+                    context.violationEndPosition = context.currentPosition.advanced(by: totalAdvance + bytesPerNewline)
+                }
+                totalAdvance += bytesPerNewline
+            }
+
+            return (newConsecutiveNewlines, totalAdvance)
         }
-        let patternRegex = regex(pattern)
-        var fileContents = file.contents
-        for violationRange in violatingRanges.reversed() {
-            fileContents = patternRegex.stringByReplacingMatches(
-                in: fileContents,
-                options: [],
-                range: violationRange,
-                withTemplate: "$2"
+    }
+
+    final class Rewriter: ViolationsSyntaxRewriter<VerticalWhitespaceClosingBracesConfiguration> {
```

Comment:
The rewriter is currently not used at all since the visitor does the replacement already and the rule doesn't specify `explicitRewriter: true`.